.github/workflows/conformance-kubespray.yaml

```
@@ -0,0 +1,251 @@
+    env:
+      job_name: "Installation and Connectivity Test"
+    steps:
+      - name: Collect Workflow Telemetry
+        uses: catchpoint/workflow-telemetry-action@94c3c3d9567a0205de6da68a76c428ce4e769af1 # v2.0.0
+        with:
+          comment_on_pr: false
+
+      - name: Checkout context ref (trusted)
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          ref: ${{ inputs.context-ref || github.sha }}
+          persist-credentials: false
+
+      - name: Set Environment Variables
+        uses: ./.github/actions/set-env-variables
+
+      - name: Get Cilium's default values
+        id: default_vars
+        uses: ./.github/actions/helm-default
+        with:
+          image-tag: ${{ inputs.SHA || github.sha }}
+          chart-dir: ./untrusted/install/kubernetes/cilium
+
+      - name: Set up Python
+        uses: actions/setup-python@a26af69be951a213d495a4c3e4e4022e16d87065 # v5.6.0
+        with:
+          python-version: '3.10'
+
+      - name: Checkout Kubespray
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          repository: kubernetes-sigs/kubespray
+          ref: ${{ env.KUBESPRAY_SHA }}
+          path: kubespray
+          fetch-depth: 1
+          persist-credentials: false
+
+      - name: Setup Kubespray
+        run: |
+          python -m venv kubespray-venv
+          source kubespray-venv/bin/activate
+          cd kubespray
+          pip install -r requirements.txt
+
+      - name: Configure inventory
+        run: |
+          cd kubespray/
+          cp -rfp inventory/sample inventory/mycluster
+
+          cat > inventory/mycluster/inventory.ini << EOF
+          [all]
+          localhost ansible_connection=local
+
+          [kube_control_plane]
+          localhost
+
+          [etcd]
+          localhost
+
+          [kube_node]
+          localhost
+
+          [k8s_cluster:children]
+          kube_control_plane
+          kube_node
+          EOF
+
+      - name: Deploy Kubernetes cluster
+        run: |
+          source kubespray-venv/bin/activate
+          cd kubespray/
+          ansible-playbook -i inventory/mycluster/inventory.ini cluster.yml -b -v \
+            -e '{"container_manager":"docker","docker_package_info":{"pkgs":[]}}' \
+            -e '{"kube_network_plugin":"cni","kube_owner":"root"}'
+
+          mkdir -p $HOME/.kube
+          sudo cp -i /etc/kubernetes/admin.conf $HOME/.kube/config
+          sudo chown "$(id -u)":"$(id -g)" $HOME/.kube/config
+
+          kubectl wait --for=condition=Ready pods --all -n kube-system --timeout=300s
+
+      - name: Set up job variables
+        id: vars
+        run: |
+          CILIUM_INSTALL_DEFAULTS="${{ steps.default_vars.outputs.cilium_install_defaults }} \
+            --helm-set=cluster.name=${{ env.clusterName }}"
+          CONNECTIVITY_TEST_DEFAULTS="--test-concurrency=5 \
+            --test='!/pod-to-world'"
```

Small nit, this should help with tracking failures in Cilium CI dashboards:
```suggestion
            --log-code-owners --code-owners=${CILIUM_CLI_CODE_OWNERS_PATHS} \
            --exclude-code-owners=${CILIUM_CLI_EXCLUDE_OWNERS} \
            --test='!/pod-to-world'"
```

---

done